packages/experiments/src/components/UnifiedPicker/hooks/useFloatingSuggestionItems.ts

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+
+export interface IUseFloatingSuggestionItems<T> {
+  focusItemIndex: number;
+  setFocusItemIndex: React.Dispatch<React.SetStateAction<number>>;
+  suggestionItems: T[];
+  setSuggestionItems: React.Dispatch<React.SetStateAction<T[]>>;
+  isSuggestionsShown: boolean;
+  showPicker: (show: boolean) => void;
+  selectNextSuggestion: () => void;
+  selectPreviousSuggestion: () => void;
+  getFocusedSuggestion: () => T;
+  hasSuggestionSelected: () => void;
+  removeSuggestion: (index: number) => void;
+}
+
+export const useFloatingSuggestionItems = <T extends {}>(
+  floatingSuggestionItems: T[],
+  focusSuggestionIndex?: number,
+  isSuggestionsVisible?: boolean,
+) => {
+  const [focusItemIndex, setFocusItemIndex] = React.useState(focusSuggestionIndex || -1);
```

Comment:
I think that it might make sense to put all of these state changes into callbacks, that way there is the potential in the future to have more control over how the states are actually updated.

Reply:
Yes, will do that, I'm just focusing on features first, will optimize once features are complete.